system/system_monitor/reader/hdd_reader/hdd_reader.cpp

```diff
@@ -385,11 +385,11 @@ int get_nvme_smart_data(int fd, HddInfo * info)
   // is from 1 to 1,000, three indicates that the number of 512 byte data
   // units written is from 2,001 to 3,000)
   info->is_valid_total_data_written_ = true;
-  info->total_data_written_ = *(reinterpret_cast<uint64_t *>(&data[48]));
+  std::memcpy(&info->total_data_written_, &data[48], sizeof(uint64_t));
```

Comment:
```suggestion
  std::memcpy(&info->total_data_written_, &data[48], sizeof(data[0]));
```

Reply:
`sizeof(data[0]) == 1` no?

Reply:
nope, it's equal to [array element size. Click to see](https://gcc.godbolt.org/z/a3refGsaK)

Reply:
But `data` is a buffer for `unsignes char`:

https://github.com/autowarefoundation/autoware.universe/blob/85b471e529b6f00e0c976988825935c1f8f6ed26/system/system_monitor/reader/hdd_reader/hdd_reader.cpp#L357

Reply:
oo, sorry didn't notice that. So My sugestion is to depend the size on total_data_written_:
``` 
std::memcpy(&info->total_data_written_, &data[48], sizeof(info->total_data_written_,));
```